### ISSUE TITLE
Create <Text /> component

### ISSUE THREAD

**Issue by darkwindcz** [kiwicom/orbit-components#91]
**Description**: The most basic component for rendering text blocks.

### Visual look:
![body-text](https://user-images.githubusercontent.com/6851247/37559090-76129e4c-2a28-11e8-9531-0c615dc705e7.png)

We should be able to render all of these styles with these parameters:

Parameter | Values | Default
-- | -- | --
Style | Primary, Secondary, Attention | Primary
Size | Large, Normal, Small | Normal
Weight | bold, normal | normal
Italic | [boolean] | false
Uppercase | [boolean] | false
Element | p, span, div, (anything else?) | p

## Design tokens

**For all configurations:**
  font-family: $fontFamily
  line-height: $lineHeightText

**Based on selected configuration:**

Parameter:Value | property: $token
-- | --
Style:Primary [optional] | color: $colorTextPrimary
Style:Secondary [optional] | color: $colorTextSecondary
Style:Attention [optional] | color: $colorTextAttention
Size:Large [optional] | font-size: $fontSizeTextLarge
Size:Normal [optional] | font-size: $fontSizeTextNormal
Size:Small [optional] | font-size: $fontSizeTextSmall
Weight:normal [optional] | font-weight: $fontWeightNormal
Weight:bold [optional] | font-weight: $fontWeightBold

Italic and uppercase don't need tokens as they aren’t expressed with numeric values.

**Possible examples of configuration:**

    <Text>A) <strong>We're here for you.</strong> First, let's narrow down your request.</Text>
    <Text Element="div" Italic>B) We're here for you. First, let's narrow down your request.</Text>

**Rendered HTML on those configurations:**

    <p>A) <strong>We're here for you.</strong> First, let's narrow down your request.</p>
    <div><em>B) We're here for you.</strong> First, let's narrow down your request.</em></div>

Note: Please take this just as my component interpretation for better understanding how I meant which param. I am totally not sure about `<strong>` though, but it should be possible to make just part of the text bold (or link in other cases). I don’t have to include “real” code, just give me a feedback how you want to specify it and I can do it your way. I won’t be able to add these code examples to more complicated components anyway.

## Some inspiration from others React components for solving text component:
We probably don't need to reinvent the wheel, a lot of other teams solved these basics already. I have a good collection of links with some inspiration from other design systems so I can include for every common component so we have something to study and inspire from 🙂​Hope it helps a little! I think that it should be possible to check their Github repositories, some of them can have them public.

https://pricelinelabs.github.io/design-system/Text
http://grommet.io/dxc/docs/paragraph (check Examples button on the right top)
https://cloudflare.github.io/cf-ui/#cf-component-text
http://design-system.pluralsight.com/components/text/
https://seek-oss.github.io/seek-style-guide/text ( + subcomponent for strong, etc.: https://seek-oss.github.io/seek-style-guide/strong)
https://backpack.github.io/components/web/text

**Comment by git-toni** [comment]
We've got `Typography`, should we rename it to `Text`?

**Comment by darkwindcz** [comment]
Yes, if it's possible. It will be synced not only with Orbit UI for Sketch but mostly with names of design tokens (colorTextPrimary, colorTextSecondary, etc.) If I am not mistaken, now `Typography` covers all texts from body texts and headlines to any other text you can imagine. 

I explained why we should have this separated in the original Dropbox Paper document, I am copying it here too:

_Why have a specific component for text and a second one for the heading?_
_Because in the design system, they both have a totally different purpose (and as you can see in specifications, also very different possibilities of configuration)._

_Also, in the code, reading will be much easier, for example:_
```
<Heading Element=”h1” Size=”Size 1”>Very important heading</Heading>
​​<Text>Loooong paragraphs of text</Text>
​​<Text>Another loooong paragraphs of text</Text>
​​<Heading Element=”h2” Size=”Size 2”>Not so important header</Heading>
​​<Text>Just another paragraph of text</Text>
```

Hope it make sense :)

**Comment by darkwindcz** [comment]
Actually, I was thinking if we shouldn't have all the Orbit components prefixed with "Orbit", e.g. <OrbitText />, <OrbitHeadline />, <OrbitTable />, <OrbitWhatever />. If we do it, it will be totally clear which components in the code are from design system and which are specific for every project. It also probably solve the issue with component <Button /> from old Kiwi repo and new <Button /> from Orbit repo (not sure if that's even issue though). 

What do you think about it? @jaroslav-kubicek @michalsanger @Fallion @infi-pc

**Comment by michalsanger** [comment]
Does other component libraries have naming like this?

**Comment by Fallion** [comment]
Developers know (or should know/ can ask their team leads) how to rename imports when using them in code.

I don't think it's a good practice to prefix stuff like this. You can already see it on the import level where the components are coming from.

Example:
`import { Input } from 'frontendcomponents'`
`import { Input as OrbitInput } from '@kiwicom/orbit-components'`

What the alternative would look like for everything else:
`import { OrbitInput } from '@kiwicom/orbit-components'`

I can't think of any library that does this.

**Comment by darkwindcz** [comment]
Not majority but there are few, yes. For example [Skyscanner with "Bpk" (Backpack)](https://backpack.github.io/components/web/buttons) or [Hubsport prefixes UI for every component](https://canvas.hubspot.com/components/buttons-links). Others that have component libraries based on CSS prefix their classes with something (Salesforce with "slds", etc.)

My idea behind it was to make clear what's from Orbit and what's not. @Fallion's example is solving this issue too, we just need to tell devs to rename imports when importing something. I believe that this can really help to have better clarity.

**Comment by Fallion** [comment]
The issue is that it gets pretty long and it's extra typing that will not be necessary half the time.

Example:
`import { OrbitButton, OrbitTypography, OrbitInput, OrbitTable } from '@kiwicom/orbit-components'`

**Comment by Fallion** [comment]
I would check with the devs whether they run into this issue or not, because if they are then the team leads should solve this knowledge issue.

**Comment by FilipMessa** [comment]
I don't like this prefix, always when I I need to know what component is from what I can just look on the import.

**Comment by git-toni** [comment]
@jaroslav-kubicek Any final decision on this?
